### PR TITLE
Implement metadata_cols and feature_cols properties in Dataset

### LINKED ISSUE #2: DataFrameWithInfo features columns could be a subset of metadata columns
At the moment `DataFrameWithInfo` takes as parameters `metadata_cols` (tuple, i.e. the columns of the dataframe to be used as metadata) and `metadata_as_features` (bool, i.e. whether to consider all the metadata columns as features). It could be useful to set only a subset of the metadata columns as features, or even to exclude some columns both from the metadata and the features sets.

To accomplish these goals, `DataFrameWithInfo` could accept both `metadata_cols` and `feature_cols` parameters as tuples of strings enumerating the metadata and the feature columns, respectively. I suggest `feature_cols` to be an optional parameter, by default equal to `None`, meaning that all the columns but the metadata columns are features columns.

### LINKED ISSUE #44: Dataset property: metadata_cols and feature_cols
metadata_cols [2a] feature_cols [2a] as properties —> these two sets can be overlapping

### PR DESCRIPTION
**Pull request by alessiamarcolini**
fixes #2 
fixes #44

### CODE REVIEW

File: src/trousse/dataset.py [hunk 1]
```diff
@@ -232,9 +232,12 @@ def __init__(
         ----------
         metadata_cols: Tuple[str], optional
             Tuple with the name of the columns that have metadata information related to
-            the sample. If ``metadata_as_features`` is set to False, these will not
-            be considered in following analysis or by "med_exam_col_list" property.
+            the sample.
             Default set to ()
+        feature_cols: Tuple[str], optional
+            Tuple with the name of the columns that contains sample features.
+            Default is None, meaning the all the columns but the metadata_cols will be
```
Comment (lorenz-gorini) [inline src/trousse/dataset.py:239]:
```suggestion
            Default is None, meaning that all the columns but ``metadata_cols`` will be
```

File: src/trousse/dataset.py [hunk 2]
```diff
@@ -608,7 +627,7 @@ def show_columns_type(self, col_list: Tuple[str] = None) -> None:
             Tuple of the name of columns that should be considered.
             If set to None, every column of ``df`` attribute will be considered.
```
Comment (lorenz-gorini) [inline src/trousse/dataset.py:628]:
```suggestion
            If set to None, only the columns in ``self.feature_cols`` property will be considered.
```

**Review by lorenz-gorini — CHANGES_REQUESTED** [review]
Minor comments. I got to admit that it is a bit hard from the PR to review every change and it is harder to understand if further changes are needed, but everything looks good in my opinion

**Comment by alessiamarcolini** [comment]
@lorenz-gorini you could try to look at the PR commit-by-commit and not the entire diff at once

**Review by lorenz-gorini — APPROVED** [review]

**Pull request merged**